packages/block-editor/src/store/selectors.native.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Internal dependencies
+ */
+import {
+	getBlockHierarchyRootClientId,
+	getBlockOrder,
+	isBlockSelected,
+	isBlockInsertionPointVisible,
+} from './selectors.js';
+
+export * from './selectors.js';
+
+export function shouldAllowSpaceForFloatingToolbar( state, clientId ) {
```

Comment:
Having trouble understanding what this function does and why we need it and why isn't it enough to only depend on "shouldShowFloatingToolbar"?

Reply:
I do still need to add documentation for this and for `shouldShowFloatingToolbar` but to answer the question here:
This function will notify if this block needs to show the floating toolbar or if any of its children in the first position are showing the floating toolbar or allowing space for the toolbar. The first position becomes important because if the floating toolbar is in the first position if might be flowing outside of its parent in which case the parent needs to allow the detection of touches in the same area as the child. 

This bubbles the tappable space from a potentially deeply nested block all the way up the chain to the main block list